src/components/search-bar/search-bar.jsx

```diff
@@ -37,11 +59,31 @@ export default class SearchBar extends React.Component {
       }
     }
     if (this.props.onSearch) {
-      const query = url ? {url} : null;
-      this.props.onSearch(query);
+      const query = url ? {url} : {url: null};
+      this.urlSearch(query);
     }
   }
 
+  /**
+   * Updates query state with url information
+   *
+   * @param {Object} urlQuery
+   */
+  urlSearch (urlQuery) {
+    const query = Object.assign({}, this.state.query, urlQuery);
+    this.setState({query});
```

Comment:
If we are going to have a query object in the state (see comment in constructor), this should use an update function instead of merging ahead of time:

```js
this.setState(state => ({query: Object.assign(state.query, urlQuery)}));
```